superset/connectors/sqla/models.py

```diff
@@ -167,15 +167,26 @@ def is_temporal(self) -> bool:
             self.type, utils.DbColumnType.TEMPORAL
         )
 
-    def get_sqla_col(self, label: Optional[str] = None) -> Column:
-        label = label or self.column_name
+    def get_sqla_col(
```

Comment:
I’m kind of surprised that SQLAlchemy isn’t dealing with the labels correctly. Are we not constructing the SQL query in the appropriate manner?

Reply:
Normally the dialect should do this, but apparently the BQ dialect isn't handling this properly (it does it correctly for all other operators except LIKE).

Reply:
@villebro do you think there is merit in trying to fix the BQ DB-API instead?

Reply:
I thought about it, but laziness got the best of me. But why not, let me poke at it real quick and see if it can be fixed there.

Reply:
I've played with this in the past by forcing `render_label_as_label=None` when calling the `super()`. For your scenario it seems like it may need to differ based on the context. 

For example we needed to ensure that the `ORDER BY` didn't contain the alias, 

```
from sqlalchemy.engine import Engine
from sqlalchemy.sql import literal_column, select


def test_identifier_preparer(engine: Engine) -> None:
    col = literal_column("SUM(x)").label("total")
    stmt = select([col])
    assert str(stmt) == "SELECT SUM(x) AS total"
    assert str(stmt.compile(engine)) == 'SELECT SUM(x) AS "total"'


def test_visit_label(engine: Engine) -> None:
    col = literal_column("SUM(x)").label("total")
    stmt = select([col]).order_by(col)
    assert str(stmt).endswith("ORDER BY total")
    assert str(stmt.compile(engine)).endswith("ORDER BY SUM(x)")
```

Reply:
Thanks for the pointers @john-bodley . I just forked it, will hack on it this weekend.